contracts/archive/src/ShieldedToken.compact

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma language_version >= 0.15.0;
+
+/**
+ * @module ShieldedToken (archived until further notice, DO NOT USE IN PRODUCTION)
+ * @description A shielded token module.
+ *
+ * @notice This module utilizes the existing coin infrastructure of Midnight.
+ * Due to the current limitations of the network, this module should NOT be used.
+ *
+ * Some of the limitations include:
+ *
+ * - No custom spend logic. Once users receive tokens, there's no mechanism to
+ * enforce any token behaviors. This is a big issue with stable coins, for instance.
+ * Most stable coins want the ability to pause functionality and/or freeze assets from
+ * specific addresses. This is currently not possible.
+ *
+ * - Cannot guarantee proper total supply accounting. The total supply of a given token
+ * is stored in the contract state. There's nothing to prevent users from burning
+ * tokens manually by directly sending them to the burn address. This breaks the
+ * total supply accounting (and potentially many other mechanisms).
+ *
+ * @notice This module will be revisited when the Midnight network can offer solutions to these
+ * issues. Until then, the recommendation is to use unshielded tokens.
+ *
+ * @dev Future ideas to consider:
+ *
+ * - Provide a self-minting mechanism.
+ * - Enable the Shielded contract itself to transfer.
+ * - Should this be a part of the Shielded module itself or as an extension?
+ */
+module ShieldedToken { // DO NOT USE IN PRODUCTION!
+  import CompactStandardLibrary;
+  import "../../node_modules/@openzeppelin-midnight/utils/src/Utils" prefix Utils_;
+
+  // Public state
+  export ledger _counter: Counter;
+  export ledger _nonce: Bytes<32>;
+  export ledger _totalSupply: Uint<64>;
```

Comment:
Non-blocking: Curious why `_totalSupply` is represented as a Uint<64> value in the shielded impl vs Uint<128> in the unshielded impl

Reply:
Good question! [CoinInfo](https://docs.midnight.network/develop/reference/midnight-api/ledger/type-aliases/CoinInfo) and [QualifiedCoinInfo](https://docs.midnight.network/develop/reference/midnight-api/ledger/type-aliases/QualifiedCoinInfo) => "Bounded to be a non-negative 64-bit integer" (in the `value` description)

So we can't really do anything about that^. And since we're in this weird in-between of unshielded ERC-ish standards with Uint<256> and the early max-uint fluctuations, Uint<128>, to me, was the safe middle ground. There are arguments for both waiting for Uint<256> (or something similar) or bringing it down to 64 bit for size parity with shielded

Reply:
Gotcha, thanks for pointing that out!